### PR TITLE
math-magicans add calculator UI

### PR DESCRIPTION
**Pull request by petushka1**
- add calculator component
- create buttons Array
- assign props to button values
- style UI
- check and fix linters errors

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,9 @@
...
-      </header>
-    </div>
-  );
-}
-
+const App = () => (
+  <div>
+    <Calculator />
+  </div>
+);
```
Comment (AmaduKamara) [inline src/App.js]: For this project, you are required to make your components classes class-based components.

- [ ] Kindly make this component a class-based react component. The goal is for you to know both ways of creating react apps with the good old class-based and later you'll refactor all your components into functional hooks as your milestone. 👍 Kindly work on the change

File: src/components/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          keyValue={btn.name}
+          className={btn.class}
+        />
+      ))}
+    </div>
+  </div>
+
+);
+
+export default Calculator;
```
Comment (AmaduKamara) [inline src/components/Calculator.js]: Your design looks nice and matches the design given. 👍But,

- [ ] Kindly make this component a class-based react component. The goal is for you to know both ways of creating react apps with the good old class-based and later you'll refactor all your components into functional hooks as your milestone. 👍 Kindly work on the change.

File: src/components/Calculator.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      <td>%</td>
+      <td>+</td>
+    </tr>
+    <tr>
+      <td>AC</td>
+      <td>+/-</td>
+      <td>%</td>
+    </tr>
+  </tbody>
+</table> */
```
Comment (AmaduKamara) [inline src/components/Calculator.js]: - [x] Kindly remove this commented code. It's an excellent standard to remove commented codes from your codebase to make your code clean and neat. Please remove them in order for your project to be considered complete. 👍

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @petushka1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ App structure is in place
✅ Design matches the original one 

## Required Changes ♻️
_Check the comments under the review._

## Recommendation:

- [x] Kindly make your calculator interface a little bigger. It's way too tiny right now and this is supposed to be a web app and not a mobile app. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 3 above)
Comment (petushka1) [reply src/components/Calculator.js]: @AmaduKamara AmaduKamara Kindly advise how to create class-based components when eslint rejects this and requires function-based

File: src/App.js (hunk 1 above)
Comment (sja-thedude) [reply src/App.js]: - [ ] As suggested previously, kindly make sure that the component is class-based.
Example:

```javascript
import React, { Component } from 'react';
import Calculator from ./components; // Import Calculator component

class App extends Component {
  render() {
    // reference as a js object
    return {
      <Calculator />
    };
  }
}
```

File: src/components/Calculator.js (hunk 2 above)
Comment (sja-thedude) [reply src/components/Calculator.js]: - [ ] As suggested previously, Kindly make the component of this file class-based. The example shown below-

```javascript
import React, { Component } from 'react';

class Calculator extends Component {
  render() {
    // reference as a js object
    return {
      <div>
       // Calculator app design goes here
      </div>
    };
  }
}

export default Calculator; // export Calculator component
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @petushka1, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by petushka1** [comment]
[sja-thedude](https://github.com/sja-thedude) Please assist. I implemented classes initially but got linters errors:
image

Kindly advise how do I sort this out?

File: src/App.js (hunk 1 above)
Comment (Whoistolu) [reply src/App.js]: Well resolved @petushka1 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @petushka1 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**